__tests__/mempool.test.js

```diff
@@ -0,0 +1,153 @@
+        let mempoolClient = getMempoolClient()
+
+        it('should create a mempool client', () => {
+            expect(mempoolClient).toBeTruthy()
+        })
+
+        describe('configuration', () => {
+            it('should have baseUrl equal to https://mempool.space', () => {
+                expect(mempoolClient.defaults.baseURL).toEqual('https://mempool.space')
+            })
+
+            it('should have 1 response interceptor', () => {
+                expect(mempoolClient.interceptors.response.handlers).toHaveLength(1)
+            })
+        })
+
+        describe('functionality', () => {
+            it('should return status 404 when unknown api is called', async () => {
+                try {
+                    await mempoolClient.get('/api/v1/unknown')
+                } catch (error) {
+                    console.log(error)
+                    expect(error.response.status).toBe(404)
+                }
+            })
+
+            it('should return status 200 when /api/v1/fees/recommended is called', async () => {
+                expect.assertions(1)
+
+                const response = await mempoolClient.get('/api/v1/fees/recommended')
+                expect(response.status).toBe(200)
+            })
+
+            // eslint-disable-next-line max-len
+            it('should return status 200 and data.activity when /api/activity is called and MEMPOOL_RETRY_URL=https://www.boredapi.com (unknown endpoint for https://mempool.space, but known for https://www.boredapi.com)', async () => {
+                // Given
+                expect.assertions(2)
+
+                // When
+                jest.resetModules()
+                process.env.MEMPOOL_RETRY_URL = 'https://www.boredapi.com'
+                mempoolClient = require('../utils/mempool').getMempoolClient()
+
+                // Then
+                const response = await mempoolClient.get('/api/activity')
+                expect(response.data.activity).toBeTruthy()
+                expect(response.status).toBe(200)
+            })
+
+            // eslint-disable-next-line max-len
+            it('should return status 404 and no data.activity when /api/activity is called and MEMPOOL_RETRY_URL is default (unknown endpoint for https://mempool.space, but known for https://www.boredapi.com)', async () => {
+                // Given
+                expect.assertions(2)
+
+                // When
+                jest.resetModules()
+                process.env.MEMPOOL_RETRY_URL = undefined
+                mempoolClient = require('../utils/mempool').getMempoolClient()
+
+                // Then
+                try {
+                    const response = await mempoolClient.get('/api/activity')
+                } catch (error) {
+                    expect(error.response.data?.activity).toBeFalsy()
+                    expect(error.response.status).toBe(404)
+                }
+            })
+        })
+    })
```

Comment:
`// eslint-disable-next-line max-len` these two tests are using real api from https://www.boredapi.com as retry, since we envoke api that exists there but not on mempool we ensure that interceptor is calling ttps://www.boredapi.com under the hood once mempool.space fails